packages/core/src/mocks.ts

```diff
@@ -0,0 +1,43 @@
+import { ConstraintsMapping } from "./interfaces";
+
+const a : ConstraintsMapping = {
+    "__self": {
+
+    },
+    "path.to.collection" : {
+        uniqueBy: ["key.property.path", "key2"],
+        length: {
```

Comment:
Try write it in the mock, It kind of duplicates the key "path.to.collection" with "path.to.collection.length" beside it. Somehow it feels more natural to just put it inside the first one.

Reply:
maybe it can be solved with prerequisites, which may make me think, the prerequisite constraints act just as sub constraint to the current key, meaning they help to merge common paths.